lib/lotus/mailer/dsl.rb

```diff
@@ -0,0 +1,157 @@
+      #     end
+      #   end
+      #
+      #   Lotus::Mailer.configuration.root # => 'app/templates'
+      #   Articles::Show.root            # => 'app/templates'
+      #
+      # @example Custom root
+      #   require 'lotus/mailer'
+      #
+      #   module Articles
+      #     class Show
+      #       include Lotus::Mailer
+      #       root 'path/to/articles/templates'
+      #     end
+      #   end
+      #
+      #   Lotus::Mailer.configuration.root # => 'app/templates'
+      #   Articles::Show.root            # => 'path/to/articles/templates'
+      def root(value = nil)
+        if value.nil?
+          configuration.root
+        else
+          configuration.root(value)
+        end
+      end
+
+      # When a value is given, specify the relative path to the template.
+      # Otherwise, it returns the name that follows Lotus::Mailer conventions.
+      #
+      # @param format [Symbol] template format
+      # @param value [String] relative template path
+      #
+      # @return [Template] the template with the correspondent format for the mailer
+      #
+      # @since 0.1.0
+      #
+      # @example Custom template
+      #   require 'lotus/mailer'
+      #
+      #   module Articles
+      #     class Show
+      #       include Lotus::Mailer
+      #       template :json, 'articles/single_article'
+      #     end
+      #
+      #   Articles::Show.template(:json)      # 'articles'
+      #
+      # @example With nested namespace
+      #   require 'lotus/mailer'
+      #
+      #   module Frontend
+      #     Mailer = Lotus::Mailer.generate(self)
+      #
+      #     class InvoiceMailer
+      #       include Frontend::Mailer
+      #     end
+      #
+      #     module Mailers
+      #       class Invoice
+      #         include Frontend::Mailer
+      #       end
+      #
+      #       module Sessions
+      #         class New
+      #           include Frontend::Mailer
+      #         end
+      #       end
+      #     end
+      #   end
+      #
+      #   Frontend::InvoiceMailer.template       # => 'standalone_mailer'
+      #   Frontend::Mailers::Invoice.template    # => 'standalone'
+      #   Frontend::Mailers::Sessions::New.template # => 'sessions/new'
+      def template(format = nil, value = nil)
+        if value.nil?
+          if !@templates.has_key?(format)
+            @templates[format] = Rendering::TemplateName.new(name, configuration.namespace).to_s
+          else
+            @templates[format]
+          end
+        else
+          @templates[format] = Mailer::Template.new("#{ [root, value].join('/') }")
+        end
+      end
+
+      # Returns the Hash with all the templates of the mailer
+      #
+      # @return [Hash] the Hash with the templates
+      #
+      # @since 0.1.0
+      #
+      # @example fetch templates
+      #   require 'lotus/mailer'
+      #   class InvoiceMailer
+      #     include Lotus::Mailer
+      #     self.templates
```

Comment:
Is there any reason why we call this method within the context of InvoiceMailer here?

Can we do:

```
class InvoiceMailer
  include Lotus::Mailer
  template :json, 'invoices.json.erb'
  template :html, 'invoices.html.erb'
end

InvoiceMailer.templates
# => the results of that^
```

instead?